test/accordion.test.ts

```diff
@@ -0,0 +1,168 @@
+      el.open = true;
+  
+      await waitUntil(() => showHandler.calledOnce);
+      await waitUntil(() => afterShowHandler.calledOnce);
+  
+      expect(showHandler).to.have.been.calledOnce;
+      expect(afterShowHandler).to.have.been.calledOnce;
+      expect(body.hidden).to.be.false;
+    });
+  
+    it('should emit sgds-hide and sgds-after-hide when setting open = false', async () => {
+      const el = await fixture<SgdsAccordionItem>(html`
+        <sgds-accordion-item open>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+          consequat.
+        </sgds-accordion-item>
+      `);
+      const body = el.shadowRoot!.querySelector<HTMLElement>('.accordion-body')!;
+      const hideHandler = sinon.spy();
+      const afterHideHandler = sinon.spy();
+  
+      el.addEventListener('sgds-hide', hideHandler);
+      el.addEventListener('sgds-after-hide', afterHideHandler);
+      el.open = false;
+  
+      await waitUntil(() => hideHandler.calledOnce);
+      await waitUntil(() => afterHideHandler.calledOnce);
+  
+      expect(hideHandler).to.have.been.calledOnce;
+      expect(afterHideHandler).to.have.been.calledOnce;
+      expect(body.hidden).to.be.true;
+    });
+  
+    it('should not open when preventing sgds-show', async () => {
+      const el = await fixture<SgdsAccordionItem>(html`
+        <sgds-accordion-item>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+          consequat.
+        </sgds-accordion-item>
+      `);
+      const showHandler = sinon.spy((event: Event) => event.preventDefault());
+  
+      el.addEventListener('sgds-show', showHandler);
+      el.open = true;
+  
+      await waitUntil(() => showHandler.calledOnce);
+  
+      expect(showHandler).to.have.been.calledOnce;
+      expect(el.open).to.be.false;
+    });
+  
+    it('should not close when preventing sgds-hide', async () => {
+      const el = await fixture<SgdsAccordionItem>(html`
+        <sgds-accordion-item open>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+          consequat.
+        </sgds-accordion-item>
+      `);
+      const hideHandler = sinon.spy((event: Event) => event.preventDefault());
+  
+      el.addEventListener('sgds-hide', hideHandler);
+      el.open = false;
+  
+      await waitUntil(() => hideHandler.calledOnce);
+  
+      expect(hideHandler).to.have.been.calledOnce;
+      expect(el.open).to.be.true;
+    });
+  });
```

Comment:
good to add the test for the nth-type attribute, last of type and first of type

Reply:
Added in commit [75597b4](https://github.com/GovTechSG/sgds-web-component/pull/42/commits/75597b4c3606d7f64860f7a6921c4018a4881137)